conda_smithy/templates/build_steps.sh.tmpl

```diff
@@ -118,6 +118,7 @@ else
     {{ BUILD_CMD }} --recipe "${RECIPE_ROOT}" \
      -m "${CI_SUPPORT}/${CONFIG}.yaml" \
      ${EXTRA_CB_OPTIONS:-} \
+     --target-platform "${HOST_PLATFORM}" \
```

Comment:
Can we call it `--host-platform` in rattler-build? Pretty please!

Reply:
We can definitely add an alias but would you also want to change `target_platform` variable everywhere in config files / recipes? We kept with conda-build here ... but `host_platform` does make _some_ sense, too.

Reply:
Make `host_platform` the canonical thing and make `target_platform` an alias is the best option.

Reply:
That is make `target_platform` env variable and jinja variable a alias to `host_platform` when `target_platform` is not explicitly given.

Reply:
Sure what order do we want to do things in? We are talking about three PRs including this one.

Reply:
I think we should merge this, and open an issue on rattler-build for an alias.

Reply:
I made an issue for this on the smithy side: https://github.com/conda-forge/conda-smithy/issues/2072. We can revisit once rattler-build has the option.

Reply:
Also this one: https://github.com/prefix-dev/rattler-build/issues/1075

Reply:
@isuruf you ok resolving this comment for now and going ahead with this PR?